crates/hyperqueue/src/server/restore.rs

```diff
@@ -290,36 +290,39 @@ impl StateRestorer {
                         }
                     }
                 }
-                EventPayload::TaskCanceled { task_id } => {
-                    log::debug!("Replaying: TaskCanceled {task_id}");
-                    if let Some(job) = self.jobs.get_mut(&task_id.job_id()) {
-                        let task = job.tasks.get_mut(&task_id.job_task_id());
-                        if let Some(task) = task {
-                            task.state =
-                                match std::mem::replace(&mut task.state, JobTaskState::Waiting) {
-                                    JobTaskState::Running { started_data } => {
-                                        JobTaskState::Canceled {
-                                            started_data: Some(started_data),
-                                            cancelled_date: event.time,
+                EventPayload::TaskCanceled { task_ids } => {
+                    log::debug!("Replaying: TaskCanceled {task_ids:?}");
+                    for task_id in task_ids {
+                        if let Some(job) = self.jobs.get_mut(&task_id.job_id()) {
```

Comment:
As a micro-optimization, we could store the last resolved job and check if the current `task_id` has the same job ID (since this will usually be the case). But cancelling shouldn't be performance critical, so it probably isn't worth it.